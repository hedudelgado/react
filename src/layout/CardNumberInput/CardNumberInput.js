import React from 'react';
import {
  string,
  bool,
  func,
  object,
} from 'prop-types';
import classnames from 'classnames';
import './CardNumberInput.scss';
import Img from '../Img';

const CardNumberInput = ({
  field,
  sessioncamhide,
  touched,
  error,
  label,
  card: {
    src,
    alt,
  },
  ...props
}) => {
  const showError = error && touched;
  const showValid = !error && touched;
  const cssClasses = classnames({
    sessioncamhidetext: sessioncamhide,
    'error-field': showError,
  });

  return (
    <div className="wb-form-item mb2">
      {label && <label className="wb-strong" htmlFor={field.name}>{label}</label>}
      <div className="wb-form-item__field wb-form-item__field__full">
        <input
          type="tel"
          className={cssClasses}
          label={label}
          {...field}
          {...props}
        />
        {src && (
          <Img
            className="card-logo mr1"
            src={src}
            label={alt}
          />
        )}
        {showError && <span className="form-item-msg__error-icon wb-icon_disc-info-filled" />}
        {showValid && <span className="form-item-msg__info-icon wb-icon_disc-checked-fill" />}
      </div>
      {showError && <span className="form-item-msg form-item-msg--error">{error}</span>}
    </div>
  );
};

CardNumberInput.propTypes = {
  /**
    HTML placeholder attribute.
  */
  placeholder: string,
  /**
    HTML value attribute.
  */
  value: string,
  /**
    Input label. Not displayed if empty.
  */
  label: string,
  /**
    Sessioncam field masking.
  */
  sessioncamhide: bool,
  /**
    Error message. Not displayed if empty.
  */
  error: string,
  /**
    Function to be run on input change.
  */
  onChange: func,
  field: object,
  touched: bool,
  card: object,
};

CardNumberInput.defaultProps = {
  card: {},
};

export default CardNumberInput;
