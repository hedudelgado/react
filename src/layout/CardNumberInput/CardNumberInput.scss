.card-logo {
  position: absolute;
  right: 8px;
  top: 6px;
}
