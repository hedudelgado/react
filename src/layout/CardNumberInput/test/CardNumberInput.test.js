import React from 'react';
import { shallow } from 'enzyme';

import CardNumberInput from '../CardNumberInput';

describe('Input Component', () => {
  const props = {
    id: 'test',
    type: 'text',
    placeholder: 'Placeholder',
    value: 'value',
    label: 'Label',
    touched: true,
    field: {
      name: 'test',
    },
  };

  it('should render the component', () => {
    const wrapper = shallow(<CardNumberInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render label if not set', () => {
    const wrapper = shallow(<CardNumberInput {...{ ...props, label: null }} />);
    expect(wrapper).not.toContainMatchingElement('label');
  });

  it('should render the error if set', () => {
    const wrapper = shallow(<CardNumberInput {...{ ...props, error: 'error message' }} />);
    expect(wrapper).toContainMatchingElement('.error-field');
    expect(wrapper).toContainMatchingElement('.form-item-msg--error');
    expect(wrapper.find('.form-item-msg--error')).toHaveText('error message');
  });

  it('should set sessioncamhide class if set', () => {
    const wrapper = shallow(<CardNumberInput {...{ ...props, sessioncamhide: true }} />);
    expect(wrapper.find('input')).toHaveClassName('sessioncamhidetext');
  });

  it('should add img to input when src is set', () => {
    const wrapper = shallow(<CardNumberInput {...{ ...props, card: { src: 'test.png', alt: 'test' } }} />);
    expect(wrapper).toContainExactlyOneMatchingElement('.card-logo');
  });
});
