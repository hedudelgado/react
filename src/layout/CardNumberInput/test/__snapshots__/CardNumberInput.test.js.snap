// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render the component 1`] = `
<div
  className="wb-form-item mb2"
>
  <label
    className="wb-strong"
    htmlFor="test"
  >
    Label
  </label>
  <div
    className="wb-form-item__field wb-form-item__field__full"
  >
    <input
      className=""
      id="test"
      label="Label"
      name="test"
      placeholder="Placeholder"
      type="text"
      value="value"
    />
    <span
      className="form-item-msg__info-icon wb-icon_disc-checked-fill"
    />
  </div>
</div>
`;
